app/auth.go

```diff
@@ -208,6 +220,40 @@ func NewAuthManager(conf AuthManagerConfig) *MenderAuthManager {
 	return mgr
 }
 
+// handleCachedTenantToken Handle the cached tenant token if it is not the same
+// as the one in the configuration
+func handleCachedAuthorizationToken(db store.Store, serverURL, tenantToken []byte) error {
+	dbValue := append(serverURL, tenantToken...)
+	cachedToken, err := db.ReadAll(datastore.AuthTenantTokenName)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
This feels a bit opposite, the only thing which happens if this is not true is to return. Isn't it better to invert the condition, and return here, then you can put the entire next block outside instead.